cmd/gctsCreateRepository_test.go

```diff
@@ -0,0 +1,186 @@
+package cmd
+
+import (
+	"bytes"
+	"errors"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+)
+
+func TestGctsCreateRepositorySuccess(t *testing.T) {
+
+	config := gctsCreateRepositoryOptions{
+		Host:                "http://testHost.wdf.sap.corp:50000",
+		Client:              "000",
+		Repository:          "testRepo",
+		Username:            "testUser",
+		Password:            "testPassword",
+		RemoteRepositoryURL: "https://github.com/org/testRepo",
+		Role:                "SOURCE",
+		VSID:                "TST",
+	}
+
+	t.Run("creating repository locally successfull", func(t *testing.T) {
+
+		httpClient := httpMock{StatusCode: 200, ResponseBody: `{
+			"repository": {
+				"rid": "com.sap.cts.example",
+				"name": "Example repository",
+				"role": "SOURCE",
+				"type": "GIT",
+				"vsid": "GI7",
+				"status": "READY",
+				"branch": "master",
+				"url": "https://github.com/git/git",
+				"version": "1.0.1",
+				"objects": 1337,
+				"currentCommit": "f1cdb6a032c1d8187c0990b51e94e8d8bb9898b2",
+				"connection": "ssl",
+				"config": [
+					{
+						"key": "CLIENT_VCS_URI",
+						"value": "git@github.wdf.sap.corp/example.git"
```

Comment:
Once upon time we had the policy not to expose sap internal urls etc. Not sure if that policy is still "active". Anyway I would like to suggest to switch to e.g. `example.org` or e.g. `example.sap.com`. With that we make also clear that we don't rely on internal systems.

Applies also elsewhere below.